### ISSUE TITLE
KWASM memory proofs

### ISSUE THREAD

**Issue by yzhang90** [kframework/kore#1072]
spec:
https://github.com/kframework/wasm-semantics/blob/memory-spec/tests/proofs/memory-spec.k
(please use memory-spec branch)
command:
 ```
kprove --haskell-backend-command "/home/zy/workspace/kore/.build/kore/bin/kore-repl --repl-script /home/zy/workspace/kore/dist/kast.kscript " ~/workspace/wasm-semantics/tests/proofs/memory-spec.k -d ~/workspace/wasm-semantics/.build/defn/haskell/ --def-module KWASM-LEMMAS
```
After executing 1 step, the result config is:
```
#Ceil ( <memInst>
    <mAddr>
      MEMADDR
    </mAddr>
    <mmax>
      DotVar50
    </mmax>
    <msize>
      SIZE
    </msize>
    <mdata>
      ByteMap <| BM0 |>
    </mdata>
  </memInst> DotVar4 )
#And
...
```
Here, i'd like to write lemmas like `#Ceil(K|->V REST) => notBool (K in_keys REST)`to simplify the `#Ceil` constraint. However, the backend doesn't support applying this lemma yet (see #1045).
After executing 13 steps, it reaches the config:
```
  <wasm>
    <k>
      load { ITYPE #numBytes ( ITYPE ) ADDR modInt 4294967296 Unsigned } ~> ITYPE . store ~> DotVar2
    </k>
  ...
  </wasm>
```
and it needs to apply the following rule
```
    rule <k> load { ITYPE WIDTH EA SIGN }
          => < ITYPE > #if SIGN ==K Signed
                           #then #signedWidth(WIDTH, #getRange(DATA, EA, WIDTH))
                           #else #getRange(DATA, EA, WIDTH)
                       #fi
         ...
         </k>
         <curModIdx> CUR </curModIdx>
         <moduleInst>
           <modIdx> CUR </modIdx>
           <memAddrs> 0 |-> ADDR </memAddrs>
           ...
         </moduleInst>
         <memInst>
           <mAddr>   ADDR </mAddr>
           <msize>   SIZE </msize>
           <mdata>   DATA </mdata>
           ...
         </memInst>
      requires (EA +Int WIDTH) <=Int (SIZE *Int #pageSize())
```
Applying the above rule requires the backend to solve #1044.
I tried the PR(https://github.com/kframework/kore/pull/1063) but it still can not apply the above rule correctly.
The error is something like:
```Failed substitution coverage check!
The substitution (above, in the unifier) did not cover the axiom variables:
    Var'Unds'111:SortExportsCell{}
    Var'Unds'121:SortTypeIdsCell{}
    Var'Unds'131:SortTypesCell{}
    Var'Unds'141:SortNextTypeIdxCell{}
    Var'Unds'151:SortFuncIdsCell{}
    Var'Unds'161:SortFuncAddrsCell{}
    Var'Unds'171:SortNextFuncIdxCell{}
    Var'Unds'181:SortTabIdsCell{}
    Var'Unds'191:SortTabAddrsCell{}
    Var'Unds'201:SortMemIdsCell{}
    Var'Unds'211:SortGlobIdsCell{}
    Var'Unds'221:SortGlobalAddrsCell{}
    Var'Unds'231:SortNextGlobIdxCell{}
    VarADDR0:SortInt{}
    VarDATA:SortByteMap{}
    VarDotVar40:SortMemInstCellMap{}
    VarDotVar51:SortMmaxCell{}
    VarDotVar70:SortModuleInstCellMap{}
    VarSIZE0:SortInt{}
in the left-hand side of the axiom.
CallStack (from HasCallStack):
  error, called at src/Kore/Step/Step.hs:481:6 in kore-0.0.1.0-1PYkDqc0weFIAwvx8ImQ2c:Kore.Step.Step
```

**Comment by ttuegel** [comment]
To clarify: #1045 is good to have, #1044 is absolutely necessary. @virgil-serbanuta This should give the example you need for #1044.

**Comment by yzhang90** [comment]
We also need #1101

**Comment by yzhang90** [comment]
need #1107

**Comment by ttuegel** [comment]
Fixed by #1130.
